

```{r}
library(Seurat)
library(Signac)
library(EnsDb.Mmusculus.v79)    
library(BSgenome.Mmusculus.UCSC.mm10)
library(ggplot2)
library(future)
set.seed(1234)
library(dplyr)
```


```{r}
library(hdf5r)

# 6 samples were processed with cellrangerAggr first to normalize their depth.

F1.counts <- Read10X_h5("/home/kanako/ownCloud/Hayashi_Lab_Shared/Liang.singleCELL/BPA.BPS/R.analysis/07.11.22.CellrangerAggrF1/Aggr.F1.Both.1and2set/Cellranger/F1_6samples/outs/filtered_feature_bc_matrix.h5")

F1.fragpath <- "/home/kanako/ownCloud/Hayashi_Lab_Shared/Liang.singleCELL/BPA.BPS/R.analysis/07.11.22.CellrangerAggrF1/Aggr.F1.Both.1and2set/Cellranger/F1_6samples/outs/atac_fragments.tsv.gz"
```



```{r}
annotation <- GetGRangesFromEnsDb(ensdb = EnsDb.Mmusculus.v79)
seqlevelsStyle(annotation) <- "UCSC"
genome(annotation) <- "mm10"
```


```{r}
# create a Seurat object containing the RNA data
F1 <- CreateSeuratObject(
  counts = F1.counts$`Gene Expression`,
  assay = "RNA",
  strip.suffix = TRUE,
  names.field = 2, names.delim = "-"
)

# create ATAC assay and add it to the object
F1[["ATAC"]] <- CreateChromatinAssay(
  counts = F1.counts$Peaks,
  sep = c(":", "-"),
  fragments = F1.fragpath,
  annotation = annotation
)
```



```{r}
DefaultAssay(F1) <- "ATAC"

F1 <- NucleosomeSignal(F1)
F1 <- TSSEnrichment(F1)



F1$blacklist_fraction <- FractionCountsInRegion(
  object = F1, 
  assay = 'ATAC',
  regions = blacklist_mm10
)


F1$pct_reads_in_peaks <- F1$nFeature_ATAC / F1$nCount_ATAC * 100

#some mitochondiral genes can be found within the nucleus
DefaultAssay(F1) <- "RNA"
F1[["percent.mt"]] <- PercentageFeatureSet(F1, pattern = "^mt-")

VlnPlot(
  object = F1,
  features = c("pct_reads_in_peaks","nCount_peaks", "TSS.enrichment", "nucleosome_signal","blacklist_fraction","nCount_RNA", "nCount_ATAC","percent.mt"),
  ncol = 4,
  pt.size = 0
)

FeatureScatter(F1, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")

FeatureScatter(F1, feature1 = "nCount_ATAC", feature2 = "nFeature_ATAC")
```


```{r}
Idents(F1) <- "orig.ident"

levels(F1)

#To know cell counts per sample
table(F1@meta.data$orig.ident)

F1 <- RenameIdents(F1, '1' = 'C1_1', '2' = 'A1_1','3'='S1_1','4' = 'C1_2', '5' = 'A1_2','6'='S1_2')
levels(F1)

F1[["replicate"]] <- Idents(object = F1)
```


```{r}
F1_1 <- subset(
  x = F1,
  subset = nCount_ATAC < 150000 &
    nCount_RNA < 150000 &
    nFeature_ATAC > 1000 &
    nFeature_RNA > 1000 &
    nucleosome_signal < 2.5 &
    TSS.enrichment > 1
)
F1_1
```



```{r}
table(F1_1@meta.data$replicate) 
```

```{r}
DefaultAssay(F1_1) <- "RNA"
F1_1 <- NormalizeData(F1_1, normalization.method = "LogNormalize", scale.factor = 10000)
F1_1 <- FindVariableFeatures(F1_1, selection.method = "vst", nfeatures = 2000)

F1_1 <- ScaleData(F1_1, verbose = FALSE)
F1_1 <- RunPCA(F1_1, npcs = 100, verbose = FALSE) 

F1_1 <- JackStraw(F1_1, num.replicate = 100, dims = 100)
F1_1 <- ScoreJackStraw(F1_1, dims = 1:100)
JackStrawPlot(F1_1, dims = 41:60)  ##select 46
```


```{r}
# batch correction

library(harmony)
F1_2 <- RunHarmony(
object = F1_1,
group.by.vars = 'orig.ident',
reduction = 'pca',
assay.use = 'RNA',
project.dim = FALSE,
reduction.save = "harmony_rna"
)


### a umap based on harmony_rna can be generated
## clustering
DefaultAssay(F1_2) <- "RNA"
F1_2.1 <- FindNeighbors(F1_2, reduction = "harmony_rna", dims = 1:45)
F1_2.1 <- FindClusters(F1_2.1, resolution = 0.02) 
F1_2.1 <- RunUMAP(F1_2.1, reduction = "harmony_rna",reduction.name = "umap_rna", dims = 1:45)  ## generate UMAP based on harmony_rna


DimPlot(F1, reduction = "umap_rna", split.by = "replicate")+FontSize(x.title = 8, y.title = 8)+theme(axis.text = element_text(size = 8))+theme(strip.text = element_text(size=8, face="bold"))+ ggtitle("RNA")  
ggsave("F1.6Sam.HarmonyRNA.UMAP.replicate.tiff", dpi=300, width = 5, height = 2.5)

DimPlot(F1_2.1, reduction = "umap_rna", group.by = "seurat_clusters", label = FALSE, repel = TRUE, ncol = 2)+FontSize(x.title = 8, y.title = 8)+theme(axis.text = element_text(size = 8))+theme(strip.text = element_text(size=8, face="bold"))
ggsave("F1.6Sam.HarmonyRNA.UMAP.combined2.tiff", dpi=300, width = 3, height = 2.5)
```



```{r}
F1$replicate <- factor(x = F1$replicate, levels = c('C1_1','C1_2','A1_1','A1_2',"S1_1","S1_2"))
levels(F1)
```






```{r}


DefaultAssay(F1_2.1) <- "ATAC"

F1_2.2 <- RunTFIDF(F1_2.1)
F1_2.2 <- FindTopFeatures(F1_2.2, min.cutoff = 'q0')
F1_2.2 <- RunSVD(F1_2.2)

DepthCor(F1_2.2, n=50)

F1_2.2 <-  RunHarmony(
object = F1_2.2,
group.by.vars = 'orig.ident',
reduction = 'lsi',
assay.use = 'peaks',
project.dim = FALSE,
reduction.save = "harmony_peaks"
)

F1 <- RunUMAP(F1, reduction = 'harmony_peaks', dims = 2:50, reduction.name = "umap.peaks", reduction.key = "peaksUMAP_")

DimPlot(F1, reduction = "umap.peaks", split.by = "replicate", na.value = "grey50", cols.highlight = "#DE2D26")+FontSize(x.title = 8, y.title = 8)+theme(axis.text = element_text(size = 8))+theme(strip.text = element_text(size=8, face="bold"))+ggtitle("ATAC")
ggsave("F1.6Sam.HarmonyATAC.UMAP.replicate1.tiff", dpi=300, width = 5, height = 2.5)

DimPlot(F1, reduction = "umap.peaks", group.by = "seurat_clusters", label = FALSE, repel = TRUE, ncol = 2)+FontSize(x.title = 8, y.title = 8)+theme(axis.text = element_text(size = 8))+theme(strip.text = element_text(size=8, face="bold"))
ggsave("F1.6Sam.HarmonyATAC.UMAP.combined1.tiff", dpi=300, width = 5, height = 2.5)
```

```{r}
F1 <- FindMultiModalNeighbors(
object = F1,
reduction.list = list("harmony_rna", "harmony_peaks"),
dims.list = list(1:45, 2:50),
modality.weight.name = "RNA.weight",
verbose = TRUE
)

F1 <- RunUMAP(
object = F1,
nn.name = "weighted.nn",
reduction.name = "wnn.umap",
assay = "RNA",
verbose = TRUE
)

F1 <- FindClusters(F1, graph.name = 'wsnn', resolution = 0.75)  #select 0.01



DimPlot(F1, reduction = "wnn.umap", group.by = "seurat_clusters", sizes.highlight = 1, label = TRUE, label.size = 4, repel = TRUE) + ggtitle("WNN")

DimPlot(F1, reduction = "wnn.umap", split.by = 'replicate',na.value = "grey50", cols.highlight = "#DE2D26")+FontSize(x.title = 8, y.title = 8)+theme(axis.text = element_text(size = 8))+theme(strip.text = element_text(size=8, face="bold"))


```


```{r}
F1.allclustermarkers <- FindAllMarkers(F1, test.use = "wilcox", only.pos = TRUE)
write.csv(F1.allclustermarkers, "F1.allclustermarkers.csv")
```




```{r}
DefaultAssay(F1_2.3) <- "RNA"
data("F1_2.3")
cd_genes <- c("Dazl",'Ddx4',"Sycp1","Stra8","Zbtb16","Inhba","Dcn","Igf1","Lhx9","Pdgfra", "Clu","Amh","Sox9","Rhox8",'Aard',"Acta2","Myh11","Myl6","Pdgfrb","Apoe","Dab2","Adgre1","Vwf","Tie1","Tek")
library(ggplot2)
DotPlot_scCustom(F1.final.wnn, features = cd_genes, x_lab_rotate = TRUE, flip_axes = FALSE) + xlab('Cluster') +  ylab('Gene') #+ scale_color_gradientn(colours = viridis::viridis(20), limits = c(0,2), oob = scales::squish, name = 'log2 (count + 1)')
```





```{r}
DefaultAssay(F1) <- "RNA"
# Germ cell marker  OCT4=Pou5f1
VlnPlot(F1, c("Pou5f1",'Ddx4',"Stra8","Dazl","Kit","Sall4","Zbtb16","Lin28","Id4"))

```

```{r}
# Sertoli cell markers
DefaultAssay(F1_2.3) <- "RNA"
VlnPlot(F1, c("Sox9","Rhox8",'Aard',"Amhr2","Clu","Ctsl"))   
# Sertoli cells
```

```{r}
VlnPlot(F1, c("Il7r","Cd52")) 
# Innate Lymphoid
```

```{r}
VlnPlot(F1, c("Vwf","Tie1","Tek","Cyp17a1","Cyp11a1","Hsd17b3")) 
#endothelial cells
#Leydig cells
```



```{r}
VlnPlot(F1, c("Rgs5","Itga1")) #pericyte
```


```{r}
VlnPlot(F1, c("Inhba","Dcn","Igf1","Lhx9","Pdgfra")) # stroma cells
```

```{r}
FeaturePlot(F1, c("Acta2","Myh11","Myl6"), reduction = "wnn.umap") #PTM
```


```{r}
FeaturePlot(F1, c("Adgre1","Apoe"), reduction = "wnn.umap") #macrophages
```


```{r}
levels(F1)
Idents(F1) <- "seurat_clusters"

new.cluster.ids <- c("Germ","stroma","Germ","Germ","Germ","Sertoli","Germ","Sertoli","stroma","Pericyte","Myoid", "Sertoli","Sertoli","stroma","stroma","Sertoli","Sertoli","stroma","Germ","Innate Lymphoid","stroma","stroma","stroma","Endothelial","Macrophage","Leydig","stroma")
names(new.cluster.ids) <- levels(F1)
F1 <- RenameIdents(F1, new.cluster.ids)

levels(F1)

#add to metadata
F1[["renamed.cluster.id"]] <- Idents(object = F1)
```


```{r}
clustermarkers <- FindALLMarkers(object=F1, only.pos = T, test.use = "wilcox")
write.csv(clustermarkers, "clustermarkers.csv")
```


```{r}
library(tidyverse)
library(patchwork)
library(viridis)
library(Seurat)
library(scCustomize)
library(qs)
library(ggplot2)
DimPlot(F1, reduction = "wnn.umap", group.by = "renamed.cluster.id", label = TRUE, label.size = 4, repel = TRUE) + FontSize(x.title = 12, y.title = 12)+theme(axis.text = element_text(size = 8))+theme(strip.text = element_text(size=8, face="bold"))+NoLegend()+Blank_Theme()

ggsave("F1.6Sam.UMAP.harmony.WNN.cluster.tiff", dpi=300,width = 4, height = 4)

DimPlot(F1, reduction = "wnn.umap", split.by = "replicate", label = FALSE, label.size = 4, repel = TRUE) +FontSize(x.title = 8, y.title = 8)+theme(axis.text = element_text(size = 8))+theme(strip.text = element_text(size=8, face="bold"))+ggtitle("WNN")+NoLegend()

ggsave("F1.6Sam.UMAP.harmony.WNN.replicate.tiff",  dpi=300, width = 5, height = 2.5)
```




```{r}
library(scCustomize)

df <- read.csv("markers for dotplot.csv")
df <- df[,2]
DefaultAssay(F1.final.wnn) <-"RNA"
DotPlot_scCustom(F1.final.wnn, features = df, x_lab_rotate = T, flip_axes = F, dot.scale=8) + xlab('Cluster') +  ylab('Gene')+theme(axis.title.x = element_blank(), axis.title.y = element_blank())+theme(axis.text.x = element_text(color='black', size=16))+theme(legend.title = element_text(size=0.5)+theme(legend.text = element_text(size = 1)))+theme(axis.line.x = element_line(size=0.55))+theme(axis.line.y = element_line(size=0.55))+theme(axis.ticks.x = element_line(size=0.55))+theme(axis.ticks.y = element_line(size = 0.55))+theme(legend.key.size = unit(0.5,"cm"))+theme(axis.text.y = element_text(color='black', size=20))

ggsave("F1 cluster makers.tiff", width = 15, height = 4.5, dpi = 300, bg = "white")
```



```{r}
DefaultAssay(F1.final.wnn)<-"ATAC"


F1.atac.cluster.markers <- FindAllMarkers(F1.final.wnn, only.pos = T, test.use = "LR")
write.csv(F1.atac.cluster.markers, "F1.ATAC.cluster.markers.csv")


F1.marker.name <- rownames(F1.atac.cluster.markers)
closest_genes_peaks.name <- ClosestFeature(F1.final.wnn, regions = F1.marker.name)
write.csv(closest_genes_peaks.name, "closest_genes_.peaks.name.csv")

F1.atac.cluster.markers <- read.csv("F1.ATAC.cluster.markers.csv")
F1.marker.name <- read.csv("closest_genes_.peaks.name.csv")
F1.ATAC.cluster.markers.Final <- merge(F1.atac.cluster.markers, F1.marker.name, by="query_region")
write.csv(F1.ATAC.cluster.markers.Final, "F1.ATAC.cluster.markers.Final.csv")
```

```{r}
saveRDS(F1.final.wnn, "F1.final.wnn.rds")
```

```{r}
library(ggplot2)
library(scCustomize)

p1<-CoveragePlot(F1.final.wnn, region=c("chr13-112651839-112652773","chr12-109436603-109437305","chr15-52151268-52152117","chr18-61058363-61059135","chr1-82560098-82560980","chr9-44986029-44986953","chr1-194927546-194928444","chr9-120054711-120055615","chr9-58012646-58013559"),ncol=9,peaks = FALSE, annotation = F)+NoLegend()+theme(axis.title.x = element_blank(), axis.title.y = element_blank())+theme(axis.text = element_text(color='black', size=3))+theme(legend.key.size = unit(0.3, 'cm'))+theme(legend.title = element_text(size = 3))+theme(axis.line.x = element_line(size=0.55))+theme(axis.line.y = element_line(size=0.55))+theme(axis.ticks.x = element_line(size=0.55))+theme(axis.ticks.y = element_line(size = 0.55))

p1<-p1+ theme(strip.text.y  = element_blank(),strip.background = element_blank())
p1

ggsave("F1 cluster atac makers.tiff", width = 20, height = 6, dpi = 300, bg = "white")

p1 <- CoveragePlot(F1.final.wnn, region="chr13-112651839-112652773",peaks = FALSE, annotation = F, scale.factor = 1500000, sep = c(":", "-"),ymax = 28)+ theme(strip.text.y.left = element_blank(),strip.background = element_blank())+theme(axis.title.x = element_blank(), axis.title.y = element_blank())+theme(strip.text.x.bottom =  element_blank(),strip.background = element_blank())+theme(axis.text.x  = element_blank())
p2 <- CoveragePlot(F1.final.wnn, region="chr12-109436603-109437305",peaks = FALSE, annotation = F,scale.factor = 1e7, sep = c(":", "-"))+ theme(strip.text.y.left = element_blank(),strip.background = element_blank())+theme(axis.title.x = element_blank(), axis.title.y = element_blank())+theme(strip.text.x.bottom =  element_blank(),strip.background = element_blank())+theme(axis.text.x  = element_blank())
p3 <- CoveragePlot(F1.final.wnn, region="chr15-52151268-52152117",peaks = FALSE, annotation = F,scale.factor = 1e7, sep = c(":", "-"))+ theme(strip.text.y.left = element_blank(),strip.background = element_blank())+theme(axis.title.x = element_blank(), axis.title.y = element_blank())+theme(strip.text.x.bottom =  element_blank(),strip.background = element_blank())+theme(axis.text.x  = element_blank())
p4 <- CoveragePlot(F1.final.wnn, region="chr18-61058363-61059135",peaks = FALSE, annotation = F,scale.factor = 1e7, sep = c(":", "-"))+ theme(strip.text.y.left = element_blank(),strip.background = element_blank())+theme(axis.title.x = element_blank(), axis.title.y = element_blank())+theme(strip.text.x.bottom =  element_blank(),strip.background = element_blank())+theme(axis.text.x  = element_blank())
p5 <- CoveragePlot(F1.final.wnn, region="chr1-82560098-82560980",peaks = FALSE, annotation = F,scale.factor = 1e7, sep = c(":", "-"))+ theme(strip.text.y.left = element_blank(),strip.background = element_blank())+theme(axis.title.x = element_blank(), axis.title.y = element_blank())+theme(strip.text.x.bottom =  element_blank(),strip.background = element_blank())+theme(axis.text.x  = element_blank())
p6 <- CoveragePlot(F1.final.wnn, region="chr9-44986029-44986953",peaks = FALSE, annotation = F,scale.factor = 1e7, sep = c(":", "-"))+ theme(strip.text.y.left = element_blank(),strip.background = element_blank())+theme(axis.title.x = element_blank(), axis.title.y = element_blank())+theme(strip.text.x.bottom =  element_blank(),strip.background = element_blank())+theme(axis.text.x  = element_blank())
p7 <- CoveragePlot(F1.final.wnn, region="chr1-194927546-194928444",peaks = FALSE, annotation = F,scale.factor = 1e7, sep = c(":", "-"))+ theme(strip.text.y.left = element_blank(),strip.background = element_blank())+theme(axis.title.x = element_blank(), axis.title.y = element_blank())+theme(strip.text.x.bottom =  element_blank(),strip.background = element_blank())+theme(axis.text.x  = element_blank())
p8 <- CoveragePlot(F1.final.wnn, region="chr9-120054711-120055615",peaks = FALSE, annotation = F,scale.factor = 1e7, sep = c(":", "-"))+ theme(strip.text.y.left = element_blank(),strip.background = element_blank())+theme(axis.title.x = element_blank(), axis.title.y = element_blank())+theme(strip.text.x.bottom =  element_blank(),strip.background = element_blank())+theme(axis.text.x  = element_blank())
p9 <- CoveragePlot(F1.final.wnn, region="chr9-58012646-58013559",peaks = FALSE, annotation = F,scale.factor = 1e7, sep = c(":", "-"))+ theme(strip.text.y.left = element_blank(),strip.background = element_blank())+theme(axis.title.x = element_blank(), axis.title.y = element_blank())+theme(strip.text.x.bottom =  element_blank(),strip.background = element_blank())+theme(axis.text.x  = element_blank())

p1


p <- cowplot::plot_grid(
   p1, p2,p3,p4,p5,p6, p7, p8, p9, ncol=9
  
)

p
ggsave("F1 cluster atac makers.tiff", width = 8, height = 6, dpi = 300, bg = "white")

```
