

```{r}
library(clusterProfiler)
library(DOSE)
library(ggplot2)
library(enrichplot)
library(org.Mm.eg.db)
library(stringr)
```

```{r}
#up-DEGs Go Terms
PsDEs <- read.csv("F1.BPA.up.csv")  
FoldChangePsDEs <- PsDEs[,2]
names(FoldChangePsDEs) <- as.character(PsDEs[,1])
FoldChangePsDEs <- sort(FoldChangePsDEs, decreasing = TRUE)

convertIDs <- function( ids, fromKey, toKey, db, ifMultiple=c( "putNA", "useFirst" ) ) {
   stopifnot( inherits( db, "AnnotationDb" ) )
   ifMultiple <- match.arg( ifMultiple )
   suppressWarnings( selRes <- AnnotationDbi::select( 
      db, keys=ids, keytype=fromKey, columns=c(fromKey,toKey) ) )
   if( ifMultiple == "putNA" ) {
      duplicatedIds <- selRes[ duplicated( selRes[,1] ), 1 ]   
      selRes <- selRes[ ! selRes[,1] %in% duplicatedIds, ] }
   return( selRes[ match( ids, selRes[,1] ), 2 ] )
}

idPsDEs <- names(FoldChangePsDEs)
PsDEs$ENTREZID <- convertIDs(idPsDEs, "SYMBOL","ENTREZID", org.Mm.eg.db)
head(PsDEs, 4)
  
  
  
AllDEs <- read.csv("All.csv")
FoldChangeAllDEs <- AllDEs[,2] 
names(FoldChangeAllDEs) <- as.character(AllDEs[,1])
FoldChangeAllDEs <- sort(FoldChangeAllDEs, decreasing = TRUE)

idAllDEs <- names(FoldChangeAllDEs)
AllDEs$ENTREZID <- convertIDs(idAllDEs, "SYMBOL","ENTREZID", org.Mm.eg.db )
head(AllDEs, 4)


GOPsDEs.BPA <- enrichGO(PsDEs$ENTREZID,   
                    universe      = AllDEs$ENTREZID,
                    OrgDb         = 'org.Mm.eg.db',
                    ont           = "BP",
                    pAdjustMethod = "BH",
                    pvalueCutoff  = 0.05,
                    qvalueCutoff  = 0.05,
                    readable      = TRUE)



a <- c("ribonucleoprotein complex biogenesis","mRNA processing","DNA repair","regulation of translation","mitotic cell cycle phase transition","ATP metabolic process","oxidative phosphorylation","intrinsic apoptotic signaling pathway","meiotic cell cycle process","autophagy","methylation","ATP metabolic process","histone modification","aerobic respiration")

p2 <- barplot(GOPsDEs.BPA, showCategory = a, font.size=12, x = "Count")+theme(axis.title = element_text(size=8))+theme(axis.text = element_text(color="black", size=11))+theme(legend.key.width = unit(0.2,"cm"))+theme(legend.key.height = unit(0.3,"cm"))+theme(legend.text = element_text(size = 8))+theme(legend.title = element_text(size = 8))+xlab("Gene Counts")+ggtitle("BPA vs CON")+theme(plot.title = element_text(size=10,face = "bold"))+theme(axis.title = element_text(size = 10))+theme(axis.text.x = element_text(angle = 30, size=8))+ theme(legend.position = c(0.85, 0.3))+scale_fill_gradientn(colors = c("#E899AC","#fde4e3")
 )+theme(axis.line.x.bottom = element_line(color = 'black'),
    axis.line.y.left   = element_blank(),
    axis.line.y.right  = element_blank(),
    axis.text.y.right  = element_blank(),
    axis.ticks.y.right = element_blank(),
    panel.border       = element_blank())+theme(panel.grid = element_blank())
p2


ggsave("Germ.BPA vs CON.GO.UP1.tiff", width=4.7, height=4.2, dpi=300, p2)
```


```{r}
library(clusterProfiler)
library(DOSE)
library(ggplot2)
library(enrichplot)
library(org.Mm.eg.db)
library(stringr)

PsDEs <- read.csv("F1.BPS.up.csv")  
FoldChangePsDEs <- PsDEs[,2]
names(FoldChangePsDEs) <- as.character(PsDEs[,1])
FoldChangePsDEs <- sort(FoldChangePsDEs, decreasing = TRUE)

convertIDs <- function( ids, fromKey, toKey, db, ifMultiple=c( "putNA", "useFirst" ) ) {
   stopifnot( inherits( db, "AnnotationDb" ) )
   ifMultiple <- match.arg( ifMultiple )
   suppressWarnings( selRes <- AnnotationDbi::select( 
      db, keys=ids, keytype=fromKey, columns=c(fromKey,toKey) ) )
   if( ifMultiple == "putNA" ) {
      duplicatedIds <- selRes[ duplicated( selRes[,1] ), 1 ]   
      selRes <- selRes[ ! selRes[,1] %in% duplicatedIds, ] }
   return( selRes[ match( ids, selRes[,1] ), 2 ] )
}

idPsDEs <- names(FoldChangePsDEs)
PsDEs$ENTREZID <- convertIDs(idPsDEs, "SYMBOL","ENTREZID", org.Mm.eg.db)
head(PsDEs, 4)
  
  
  
AllDEs <- read.csv("All.csv")
FoldChangeAllDEs <- AllDEs[,2] 
names(FoldChangeAllDEs) <- as.character(AllDEs[,1])
FoldChangeAllDEs <- sort(FoldChangeAllDEs, decreasing = TRUE)

idAllDEs <- names(FoldChangeAllDEs)
AllDEs$ENTREZID <- convertIDs(idAllDEs, "SYMBOL","ENTREZID", org.Mm.eg.db )
head(AllDEs, 4)


GOPsDEs.BP <- enrichGO(PsDEs$ENTREZID,   
                    universe      = AllDEs$ENTREZID,
                    OrgDb         = 'org.Mm.eg.db',
                    ont           = "BP",
                    pAdjustMethod = "BH",
                    pvalueCutoff  = 0.05,
                    qvalueCutoff  = 0.05,
                    readable      = TRUE)


a <- c("ribonucleoprotein complex biogenesis","mRNA processing","DNA repair","regulation of translation","mitotic cell cycle phase transition","ATP metabolic process","oxidative phosphorylation","intrinsic apoptotic signaling pathway","meiotic cell cycle process","autophagy","methylation","ATP metabolic process","histone modification","aerobic respiration")

p1 <- barplot(GOPsDEs.BP, showCategory = a, font.size=12, x = "Count")+theme(axis.title = element_text(size=8))+theme(axis.text = element_text(color="black", size=11))+theme(legend.key.width = unit(0.2,"cm"))+theme(legend.key.height = unit(0.35,"cm"))+theme(legend.text = element_text(size = 8))+theme(legend.title = element_text(size = 8))+xlab("Gene Counts")+ggtitle("BPS vs CON")+theme(plot.title = element_text(size=10,face = "bold"))+theme(axis.title = element_text(size = 10))+theme(axis.text.x = element_text(angle = 30, size=8))+ theme(legend.position = c(0.85, 0.3))+scale_fill_gradientn(colors = c("#80bcbd","#7ed2df","#d7eaf3"))+theme(axis.line.x.bottom = element_line(color = 'black'),
    axis.line.y.left   = element_blank(),
    axis.line.y.right  = element_blank(),
    axis.text.y.right  = element_blank(),
    axis.ticks.y.right = element_blank(),
    panel.border       = element_blank())+theme(panel.grid = element_blank())+scale_x_continuous(limits = c(0, 350))
p1

ggsave("Germ.BPS vs CON.GO.UP2.tiff", width=3.7, height=4.2, dpi=300, p1)



```



```{r}
#down-DEGs GO terms together
A <- read.csv("F1.BPA.down.csv")  
FoldChangePsDEs <- A[,2]
names(FoldChangePsDEs) <- as.character(A[,1])
FoldChangePsDEs <- sort(FoldChangePsDEs, decreasing = TRUE)

convertIDs <- function( ids, fromKey, toKey, db, ifMultiple=c( "putNA", "useFirst" ) ) {
   stopifnot( inherits( db, "AnnotationDb" ) )
   ifMultiple <- match.arg( ifMultiple )
   suppressWarnings( selRes <- AnnotationDbi::select( 
      db, keys=ids, keytype=fromKey, columns=c(fromKey,toKey) ) )
   if( ifMultiple == "putNA" ) {
      duplicatedIds <- selRes[ duplicated( selRes[,1] ), 1 ]   
      selRes <- selRes[ ! selRes[,1] %in% duplicatedIds, ] }
   return( selRes[ match( ids, selRes[,1] ), 2 ] )
}


idPsDEsA <- names(FoldChangePsDEs)
A$ENTREZID <- convertIDs(idPsDEsA, "SYMBOL","ENTREZID", org.Mm.eg.db)




S <- read.csv("F1.BPS.down.csv")  
FoldChangePsDEs <- S[,2]
names(FoldChangePsDEs) <- as.character(S[,1])
FoldChangePsDEs <- sort(FoldChangePsDEs, decreasing = TRUE)

convertIDs <- function( ids, fromKey, toKey, db, ifMultiple=c( "putNA", "useFirst" ) ) {
   stopifnot( inherits( db, "AnnotationDb" ) )
   ifMultiple <- match.arg( ifMultiple )
   suppressWarnings( selRes <- AnnotationDbi::select( 
      db, keys=ids, keytype=fromKey, columns=c(fromKey,toKey) ) )
   if( ifMultiple == "putNA" ) {
      duplicatedIds <- selRes[ duplicated( selRes[,1] ), 1 ]   
      selRes <- selRes[ ! selRes[,1] %in% duplicatedIds, ] }
   return( selRes[ match( ids, selRes[,1] ), 2 ] )
}

idPsDEsS <- names(FoldChangePsDEs)
S$ENTREZID <- convertIDs(idPsDEsS, "SYMBOL","ENTREZID", org.Mm.eg.db)



DATA <- list(BPA=A$ENTREZID, BPS=S$ENTREZID)

str(DATA)
down <- compareCluster(geneCluster = DATA, fun = enrichGO, ont = "BP",
                    pAdjustMethod = "BH",
                    pvalueCutoff  = 0.05,
                    qvalueCutoff  = 0.05,
                    universe= AllDEs$ENTREZID,
                    OrgDb = 'org.Mm.eg.db',
                    readable=TRUE)
down.filter <-simplify(down, cutoff = 0.7, by = "p.adjust", select_fun = min)

b <- c("response to acid chemical","response to wounding","response to mechanical stimulus","response to amino acid","transmembrane receptor protein serine/threonine kinase signaling pathway","cellular response to glucose stimulus","cellular response to hormone stimulus","cellular response to hexose stimulus","response to calcium ion","cellular response to transforming growth factor beta stimulus","transmembrane receptor protein serine/threonine kinase signaling pathway","macrophage activation","positive regulation of cytokine production")


down.filter <- enrichplot::pairwise_termsim(down.filter) 

nodeid.tbl_tree <- utils::getFromNamespace("nodeid.tbl_tree", "tidytree")
rootnode.tbl_tree <- utils::getFromNamespace("rootnode.tbl_tree", "tidytree")
offspring.tbl_tree <- utils::getFromNamespace("offspring.tbl_tree", "tidytree")
offspring.tbl_tree_item <- utils::getFromNamespace(".offspring.tbl_tree_item", "tidytree")
child.tbl_tree <- utils::getFromNamespace("child.tbl_tree", "tidytree")
parent.tbl_tree <- utils::getFromNamespace("parent.tbl_tree", "tidytree")

p3 <- treeplot(down.filter, showCategory = b, color = "p.adjust", nWords=0, hclust_method = "average", offset=5,label_format = 5, geneClusterPanel= "pie", cex_category=2, xlim=c(0,120), offset_tiplab=0.5, group_color=c("blue","#cc7a02","red","black"), pie="Count",hilight=T, nCluster=4, fontsize=7, legend_n=3, extend=0.5)+scale_fill_manual(values=c("#f282b4","#60e3d5"))+theme(legend.position =  c(.05,.85))
p3
ggsave("Go.down1.tiff", dpi = 300, width = 12, height = 7, p3)

```
```{r}
p4 <- treeplot(down.filter, showCategory = b, color = "p.adjust", nWords=0, hclust_method = "average", offset=5,label_format = 5, geneClusterPanel= "pie", cex_category=2, xlim=c(0,120), offset_tiplab=1, group_color=c("#118ab2","#333333","#ef476f","#f77f00"), pie="Count",hilight=T, nCluster=4, fontsize=8, legend_n=3, extend=0.5)+scale_fill_manual(values=c("#BAABDA","#9FDFCD"))+theme(legend.position =  c(.05,.85))
p4
ggsave("Go.down.tiff", dpi = 300, width = 12, height = 3, p4)
```
```{r}
#Down-DEGs heatmap 
library(ggplot2)
library(dplyr)

#use combined BPA and BPS DEGs total 491
#load the genes that I want to use
df<-read.csv("F1.BP overlap.down.csv", header = T)
df1<-df[,7]


df4 <- AverageExpression(
  F1.Germ,
  assays = "RNA",
  features = df1,
  return.seurat = FALSE,
  group.by = "treatment",
  add.ident = NULL,
  slot = "data",
  verbose = TRUE
)

write.csv(df4, "heatmap_F1_down.csv")
```







```{r}
# Compare BPA vs BPS
df <- read.csv("F1.BPA.BPS.csv")  
FoldChange <- df[,2]
names(FoldChange) <- as.character(df[,1])

id <- names(FoldChange)
df$ENTREZID <- convertIDs(id, "SYMBOL","ENTREZID", org.Mm.eg.db)


mydf <- data.frame(Entrez=df$ENTREZID, FC=FoldChange)
mydf <- mydf[abs(mydf$FC) > 1,]
mydf$group <- "upregulated"
mydf$group[mydf$FC < 0] <- "downregulated"





```


```{r}
BPA.S <- compareCluster(Entrez~group, data=mydf,fun = enrichGO, ont = "BP",
                    pAdjustMethod = "BH",
                    pvalueCutoff  = 0.05,
                    qvalueCutoff  = 0.05,
                    universe= AllDEs$ENTREZID,
                    OrgDb = 'org.Mm.eg.db',
                    readable=TRUE)

head(BPA.S) 

BPA.S.filter <-simplify(BPA.S, cutoff = 0.5, by = "p.adjust", select_fun = min)
write.csv(BPA.S, "GO.BPA.S.csv")
#dim(ck.filter)

write.csv(BPA.S.filter, "GO.BPA.S.filter1.csv")
```

```{r}
c <- c("ribonucleoprotein complex biogenesis","RNA splicing","oxidative phosphorylation","regulation of translation","ATP metabolic process","cell cycle phase transition","synapse organization","small GTPase mediated signal transduction","regulation of developmental growth","positive regulation of GTPase activity")


p2 <- dotplot(BPA.S.filter, showCategory = c, font.size=12, x="group")+theme(axis.title = element_text(size=12))+theme(axis.text.x = element_text(color="black", size=12,angle = 0, vjust = 0.5, hjust= ))+theme(legend.title = element_text(size=8)+theme(legend.text = element_text(size = 8)))+theme(legend.key.width = unit(0.2,"cm"))+theme(legend.key.height = unit(0.3,"cm"))+theme(legend.text = element_text(size = 8))+theme(legend.title = element_text(size = 8))+xlab("")+theme(plot.title = element_text(size=10,face = "bold"))+theme(axis.title = element_text(size = 8))+scale_x_discrete(labels=c('Down', 'Up'))
p2

ggsave("Go.BPA.S.tiff", dpi = 300, width = 4.5, height = 4)
```


```{r}
library(EnhancedVolcano)

df <- read.csv("BPA.BPS.DEs.F1.Germ.Ming.csv") 


 keyvals <- ifelse(df$FC < 0 & df$p_val_adj <0.05, 'blue',
  
      ifelse(df$FC > 1 & df$p_val_adj <0.05, 'red',
         "black"))
 
  keyvals[is.na(keyvals)] <- 'black'
  names(keyvals)[keyvals == 'red'] <- 'Up (2721)'
 
  names(keyvals)[keyvals == 'blue'] <- "Down (521)"

head(df)
  
plot1<-EnhancedVolcano(df, 

    lab = as.character(df$Gene),
        x = 'FC',
    y = 'p_val_adj',
    title = '',
   colCustom = keyvals,
    pCutoff = 0.05,
    FCcutoff = 1.4,
    pointSize = 1.0,
    labSize = 5,
    subtitle = NULL,
    legendLabSize = 14,
    legendIconSize = 5.0,
    gridlines.major = TRUE,
    gridlines.minor = FALSE,
   drawConnectors = TRUE,
    border = 'partial',
    borderWidth = 1,
    borderColour = 'black')+xlim(-4.5,4.5)+ggtitle('BPA vs BPS')
plot1

ggsave("BPA.S.DEGs.tiff", dpi = 300, width = 6, height =7)
```

