

```{r}
readRDS("/home/kanako/ownCloud/Hayashi_Lab_Shared/Liang.singleCELL/BPA.BPS/R.analysis/10.03.22.CellrangerAggrF2/Final.R.F2.Ming/F2.final.wnn.rds")
```

```{r}
F2.Germ <- subset(F2.final.wnn, idents= c("Germ"))
```

```{r}
F2.Germ <- FindMultiModalNeighbors(
object = F2.Germ,
reduction.list = list("harmony_rna", "harmony_peaks"),
dims.list = list(1:32, 2:50),
modality.weight.name = "RNA.weight",
verbose = TRUE
)


F2.Germ <- RunUMAP(
object = F2.Germ,
nn.name = "weighted.nn",
reduction.name = "wnn.umap2Germ",
assay = "RNA",
verbose = TRUE
)

F2.Germ <- FindClusters(F2.Germ, graph.name = 'wsnn', resolution = 0.5)    #  select 0.5

DimPlot(F2.Germ, reduction = "wnn.umap2Germ", label = TRUE, pt.size = 0.5)
```


```{r}
DefaultAssay(F2.Germ) <-"RNA"
FeaturePlot(F2.Germ, c("Id4","Etv5","Gfra1","Mki67"), reduction = "wnn.umap2Germ")
```


```{r}
FeaturePlot(F2.Germ, c("Lin28a","Upp1","Kit","Stra8"), reduction = "wnn.umap2Germ")
```


```{r}
FeaturePlot(F2.Germ, c("Esx1","Lhx1","Dmrtb1","Sohlh1"), reduction = "wnn.umap2Germ")
```



```{r}
VlnPlot(F2.Germ, c("Etv5","Lhx1","Erbb3","Nanos3", "Lin28a","Sohlh1","Esx1","Upp1"), ncol = 4)
```
```{r}
VlnPlot(F2.Germ, c("Id4","Stra8","Gfra1","Tspan33","Tspan8","Sox3"))
```

```{r}
VlnPlot(F2.Germ, c("Plvap","Kit","Bcl6b","Pou5f1","Neurog3"))
```

```{r}
saveRDS(F2.Germ, "F2.Germ.rds")
```

```{r}
VlnPlot(F2.Germ, c("Rhox13","Rarg","Utf1","Etv4","Ly6h","Dapk1"))
```

```{r}
F2.Germclustermarkers <- FindAllMarkers(F2.Germ, test.use = "wilcox", only.pos = TRUE)
write.csv(F2.Germclustermarkers, "F2.Germclustermarkers1.csv")
```


```{r}
Idents(F2.Germ) <-"seurat_clusters"
levels(F2.Germ)

Germ.cluster.ids <- c("SSC3","SSC2",  "Diff","Progenitor", "SSC1", "SSC4","SSC5")
names(Germ.cluster.ids) <- levels(F2.Germ)
F2.Germ <- RenameIdents(F2.Germ, Germ.cluster.ids)

levels(F2.Germ)

#add to metadata
F2.Germ[["Germ.cluster.ids"]] <- Idents(object = F2.Germ)
```


```{r}
Idents(F2.Germ) <- "Germ.cluster.ids"
levels(F2.Germ)
F2.Germ$Germ.cluster.ids <- factor(x = F2.Germ$Germ.cluster.ids, levels = c("SSC1","SSC2","SSC3","SSC4","SSC5", "Progenitor", "Diff"))
```


```{r}
DimPlot(F2.Germ, reduction = "wnn.umap2Germ", label = T, label.size = 4, pt.size = 0.5)+ NoLegend()+Blank_Theme()+scale_color_manual(values =c('#ff9a36','#31C53F','#1FA195','#28CECA',"#F68282",'#B95FBB','#D4D915'))
ggsave("F2.Germ.final.cluster.tiff", dpi=300, width = 3, height = 2.5)

```


```{r}
DimPlot(F2.Germ, reduction = "wnn.umap2Germ", label = F, label.size = 4, pt.size = 0.5, split.by = "Treatment")+Blank_Theme()+scale_color_manual(values =c('#ff9a36','#31C53F','#1FA195','#28CECA',"#F68282",'#B95FBB','#D4D915'))
ggsave("F2.Germ.final.cluster.treatment.tiff", dpi=300, width = 8, height = 2)
```





```{r}

library(monocle3)
library(SeuratWrappers)
library(Seurat)
library(ggplot2)

DefaultAssay(F2.Germ) <- "RNA"
Idents(F2.Germ)<-"Germ.cluster.ids"

F2.germ_Diet <- DietSeurat(F2.Germ, counts=TRUE, data=TRUE, graph="wsnn", dimreducs = "wnn.umap2Germ")

F2.germ_Diet[["UMAP"]] <- F2.germ_Diet[["wnn.umap2Germ"]]  ## If THE UMAP from seurat needs to be kept, it only accepts "UMAP' or "PCA" and others. As we want to use the WNN umap, we need to name it as "UMAP" to be processed in Monocle3.

F2.germ.cds <- as.cell_data_set(F2.germ_Diet)
F2.germ.cds <- cluster_cells(F2.germ.cds, reduction_method = "UMAP")

F2.germ.cds@clusters$UMAP$partitions[F2.germ.cds@clusters$UMAP$partitions == "2"] <- "1"
F2.germ.cds <- learn_graph(F2.germ.cds, use_partition = F)

F2.germ.cds <- order_cells(F2.germ.cds, reduction_method = "UMAP")

plot_cells(cds = F2.germ.cds, label_cell_groups = FALSE, color_cells_by = "ident", show_trajectory_graph = TRUE, label_branch_points = FALSE, label_leaves = FALSE, label_roots = FALSE, cell_size = 0.5)+theme(axis.title = element_text(size=8))+theme(axis.text = element_text(color='black', size=8))+theme(legend.title = element_text(size=8)+theme(legend.text = element_text(size = 8)))+theme(axis.line.x = element_line(size=0.55))+theme(axis.line.y = element_line(size=0.55))+theme(axis.ticks.x = element_line(size=0.55))+theme(axis.ticks.y = element_line(size = 0.55))+theme(legend.title = element_text(size = 8))+theme(legend.text = element_text(size = 8))+scale_color_manual(values =c('#ff9a36','#31C53F','#1FA195','#28CECA',"#F68282",'#B95FBB','#D4D915'))
ggsave("F2.Germ.pseudotime.tiff", dpi=300, width = 5.5, height = 3.5)
```
```{r}

F2.germ.cds$monocle3_pseudotime <- pseudotime(F2.germ.cds)

data.pseudo <- as.data.frame(colData(F2.germ.cds))

ggplot(data.pseudo, aes(monocle3_pseudotime, reorder(ident, monocle3_pseudotime), fill = ident, label=TRUE)) + geom_boxplot()+scale_fill_manual(values =c('#ff9a36','#31C53F','#1FA195','#28CECA',"#F68282",'#B95FBB','#D4D915'))

ggsave("F2.pseudotimeclusterorder.tiff", dpi=300, width = 5, height = 4)
```

```{r}
F2.germ.cds <- order_cells(F2.germ.cds)
plot_cells(F2.germ.cds,
           graph_label_size = 5,
           color_cells_by = "pseudotime",
           group_cells_by = "cluster",
           label_cell_groups = FALSE,
           label_groups_by_cluster=FALSE,
           label_leaves=FALSE,
           label_branch_points=FALSE,
           label_roots = TRUE,
           trajectory_graph_color = "black")+theme(axis.title = element_text(size=8))+theme(axis.text = element_text(color='black', size=10))+theme(legend.title = element_text(size=8)+theme(legend.text = element_text(size = 8)))+theme(axis.line.x = element_line(size=0.55))+theme(axis.line.y = element_line(size=0.55))+theme(axis.ticks.x = element_line(size=0.55))+theme(axis.ticks.y = element_line(size = 0.55))+theme(legend.title = element_text(size = 12))+theme(legend.text = element_text(size = 12))

ggsave("F2.Germ.pseudotime2.tiff", dpi=300, width = 5.5, height = 3.5)
```
```{r}
saveRDS(F2.Germ, "F2.Germ.rds")
saveRDS(F2.germ.cds,"F2.germ.cds.rds")
```




```{r}
library(ggplot2)
library(monocle)


F2.germ.cds <- estimate_size_factors(F2.germ.cds)
F2.germ.cds@rowRanges@elementMetadata@listData[['gene_short_name']] <- rownames(F2.germ.cds[['RNA']])


Genes.module <- c("Stra8","Ret","Kit","Upp1","Etv4","Dmc1")

rowData(F2.germ.cds)$gene_name <- rownames(F2.germ.cds)
rowData(F2.germ.cds)$gene_short_name <- rowData(F2.germ.cds)$gene_name


Genes.module_cds <- F2.germ.cds[rowData(F2.germ.cds)$gene_short_name %in% Genes.module,label_by_short_name = FALSE]
#color_cells_by ="ident", 

plot_genes_in_pseudotime(Genes.module_cds, color_cells_by ="ident", label_by_short_name=FALSE)+theme(axis.title = element_text(size=8))+theme(axis.text = element_text(color='black', size=8))+theme(legend.title = element_text(size=8)+theme(legend.text = element_text(size = 8)))+theme(axis.line.x = element_line(size=0.55))+theme(axis.line.y = element_line(size=0.55))+theme(axis.ticks.x = element_line(size=0.55))+theme(axis.ticks.y = element_line(size = 0.55))+ theme(strip.text = element_text(size=8, face="bold.italic"))
```



```{r}
DefaultAssay(F2.Germ) <- "RNA"
Idents(F2.Germ) <- "Treatment"

BPA.CON.DEs <- FindMarkers(F2.Germ, ident.1 = "A2", ident.2 = "C2", test.use = "wilcox", logfc.threshold = 0)
library(dplyr)
BPA.CON.DEs <- BPA.CON.DEs %>%
  mutate(FC = if_else(avg_log2FC>=0, 2^(BPA.CON.DEs$avg_log2FC), -2^(-(BPA.CON.DEs$avg_log2FC))))
write.csv(BPA.CON.DEs, "BPA.CON.DEs.F2.Germ.Ming.csv")


BPS.CON.DEs <- FindMarkers(F2.Germ, ident.1 = "S2", ident.2 = "C2", test.use = "wilcox", logfc.threshold = 0)
BPS.CON.DEs <- BPS.CON.DEs %>% 
  mutate(FC = if_else(avg_log2FC>=0, 2^(BPS.CON.DEs$avg_log2FC), -2^(-(BPS.CON.DEs$avg_log2FC))))
write.csv(BPS.CON.DEs, "BPS.CON.DEs.F2.Germ.Ming.csv")


BPA.BPS.DEs <- FindMarkers(F2.Germ, ident.1 = "A2", ident.2 = "S2", test.use = "wilcox", logfc.threshold = 0)
BPA.BPS.DEs <- BPA.BPS.DEs %>% 
  mutate(FC = if_else(avg_log2FC>=0, 2^(BPA.BPS.DEs$avg_log2FC), -2^(-(BPA.BPS.DEs$avg_log2FC))))
write.csv(BPA.BPS.DEs, "BPA.BPS.DEs.F2.Germ.Ming.csv")
```

```{r}
Idents(F2.Germ) <- "seurat_clusters"
levels(F2.Germ)
DimPlot(F2.Germ, reduction = "wnn.umap2Germ", label = TRUE, pt.size = 0.5, split.by = "Treatment")
```



```{r}
table(F2.Germ@meta.data$Treatment)
```



```{r}
F2.stem <- subset(F2.Germ, idents = c("0","1","4","5","6"))
F2.diff <- subset(F2.Germ, idents = c("2"))
F2.Progenitor <- subset(F2.Germ, idents = c("3"))

table(F2.stem@meta.data$Treatment)
table(F2.Progenitor@meta.data$Treatment)
table(F2.diff@meta.data$Treatment)
```

```{r}
Idents(F2.Germ) <-"seurat_clusters"
levels(F2.Germ)
```


```{r}
VlnPlot(F2.Germ, c("Dmrtb1","Sycp3","Ly6k","Tex14"))
```


```{r}
state <- c("stem","stem", "progenitor", "diff", "stem", "stem")
names(state) <- levels(F2.Germ)
F2.Germ <- RenameIdents(F2.Germ, state)

levels(F2.Germ)

#add to metadata
F2.Germ[["state"]] <- Idents(object = F2.Germ)
```


```{r}
Idents(F2.Germ) <-"Germ.cluster.ids"
DimPlot(F2.Germ, reduction = "wnn.umap2Germ", label = FALSE, pt.size = 0.5, split.by = "Treatment")+Blank_Theme()
ggsave("F2.Germ.by_treatment.tiff", dpi=300, width = 8, height = 2)
```


```{r}
saveRDS(F2.Germ, "F2.Germ.rds")
saveRDS(stem, "F2.stem.rds")
saveRDS(Progenitor, "F2.progenitor.rds")
saveRDS(diff, "F2.diff.rds")
```




```{r}
DefaultAssay(F2.Germ) <- "ATAC"

Idents(F2.Germ) <- "Treatment"
levels(F2.Germ)

BPA.CON.peaks <- FindMarkers(F2.Germ, ident.1 = "A2", ident.2 = "C2", min.pct = 0, logfc.threshold = 0, test.use = 'LR', latent.vars = 'nFeature_ATAC')
BPS.CON.peaks <- FindMarkers(F2.Germ, ident.1 = "S2", ident.2 = "C2", min.pct = 0, logfc.threshold = 0,test.use = 'LR', latent.vars = 'nFeature_ATAC')
BPA.BPS.peaks <- FindMarkers(F2.Germ, ident.1 = "A2", ident.2 = "S2", min.pct = 0, logfc.threshold = 0,test.use = 'LR', latent.vars = 'nFeature_ATAC')
write.csv(BPA.CON.peaks, "BPA.CON.peaks.csv")
write.csv(BPS.CON.peaks, "BPS.CON.peaks.csv")
write.csv(BPA.BPS.peaks, "BPA.BPS.peaks.csv")

BPA.CON.peaks.name <- rownames(BPA.CON.peaks)
closest_genes_BPA.CON.peaks.name <- ClosestFeature(F2.Germ, regions = BPA.CON.peaks.name)
write.csv(closest_genes_BPA.CON.peaks.name, "closest_genes_BPA.CON.peaks.name.F2.csv")

BPS.CON.peaks.name <- rownames(BPS.CON.peaks)
closest_genes_BPS.CON.peaks.name <- ClosestFeature(F2.Germ, regions = BPS.CON.peaks.name)
write.csv(closest_genes_BPS.CON.peaks.name, "closest_genes_BPS.CON.peaks.name.F2.csv")

BPA.BPS.peaks.name <- rownames(BPA.BPS.peaks)
closest_genes_BPA.BPS.peaks.name <- ClosestFeature(F2.Germ, regions = BPA.BPS.peaks.name)
write.csv(closest_genes_BPA.BPS.peaks.name, "closest_genes_BPA.BPS.peaks.name.F2.csv")

#merge to show all information of differential opening

BPA.CON1 <- read.csv("BPA.CON.peaks.csv")
BPA.CON2 <- read.csv("closest_genes_BPA.CON.peaks.name.F2.csv")
BPA.CON.DO <- merge(BPA.CON1, BPA.CON2, by="query_region")
write.csv(BPA.CON.DO, "BPA.CON.DO.combined.F2.Final.csv")


BPS.CON1 <- read.csv("BPS.CON.peaks.csv")
BPS.CON2 <- read.csv("closest_genes_BPS.CON.peaks.name.F3.csv")
BPS.CON.DO <- merge(BPS.CON1, BPS.CON2, by="query_region")
write.csv(BPS.CON.DO, "BPS.CON.DO.combined.F3.Final.csv")


BPA.BPS1 <- read.csv("BPA.BPS.peaks.csv")
BPA.BPS2 <- read.csv("closest_genes_BPA.BPS.peaks.name.F2.csv")
BPA.BPS.DO <- merge(BPA.BPS1, BPA.BPS2, by="query_region")
write.csv(BPA.BPS.DO, "BPA.BPS.DO.combined.F2.Final.csv")


### add fold change

BPA.CON.DO <- BPA.CON.DO %>%
  mutate(FC = if_else(avg_log2FC>=0, 2^(BPA.CON.DO$avg_log2FC), -2^(-(BPA.CON.DO$avg_log2FC))))
write.csv(BPA.CON.DO, "BPA.CON.DO.combined.F2.Final.ATAC.csv")

BPS.CON.DO <- BPS.CON.DO %>%
  mutate(FC = if_else(avg_log2FC>=0, 2^(BPS.CON.DO$avg_log2FC), -2^(-(BPS.CON.DO$avg_log2FC))))
write.csv(BPS.CON.DO, "BPS.CON.DO.combined.F2.Final.ATAC.csv")


BPA.BPS.DO <- BPA.BPS.DO %>%
  mutate(FC = if_else(avg_log2FC>=0, 2^(BPA.BPS.DO$avg_log2FC), -2^(-(BPA.BPS.DO$avg_log2FC))))
write.csv(BPA.BPS.DO, "BPA.BPS.DO.combined.F2.Final.ATAC.csv")
```




```{r}
library(motifmatchr)
library(JASPAR2020)
library(TFBSTools)
library(BSgenome.Mmusculus.UCSC.mm10)

library(Signac)
library(Seurat)
# extract position frequency matrices for the motifs
pwm <- getMatrixSet(
  x = JASPAR2020,
  opts = list(collection = "CORE", tax_group = 'vertebrates',  all_versions = FALSE)
)
head(pwm)

gr <- granges(F2.Germ)
seq_keep <- seqnames(gr) %in% seqnames(BSgenome.Mmusculus.UCSC.mm10) 
seq_keep <- as.vector(seq_keep)
feat.keep <- GRangesToString(grange = gr[seq_keep])
F2.Germ[['ATAC']] <- subset(F2.Germ[["ATAC"]], features = feat.keep)

# add motif information
F2.Germ <- AddMotifs(F2.Germ, genome = BSgenome.Mmusculus.UCSC.mm10, pfm = pwm)

head(rownames(F2.Germ[["ATAC"]]))
```
```{r}
DefaultAssay(F2.Germ)<-"ATAC"
F2.Germ <- Footprint(
  object = F2.Germ,
  motif.name = c("SP1","Klf12"),
  genome = BSgenome.Mmusculus.UCSC.mm10
)

# plot the footprint data for each group of cells
p2 <- PlotFootprint(F2.Germ, features = c("SP1","Klf12"))

p2 + patchwork::plot_layout(ncol = 1)
```


```{r}
DefaultAssay(F2.Germ)<-"ATAC"
Idents(F2.Germ) <-"Treatment"

BPA_peaks <- FindMarkers(
  object = F2.Germ,
  ident.1 = 'A2',
  ident.2 = 'C2',
  only.pos = TRUE,
  test.use = 'LR',
  min.pct = 0.05,
  latent.vars = 'nFeature_ATAC'
)


top.da.peak <- rownames(BPA_peaks[BPA_peaks$p_val < 0.005, ])

enriched.motifs <- FindMotifs(
  object = F2.Germ,
  features = top.da.peak
)


write.csv(enriched.motifs, "F2.BPA.CON.enrichmotif.csv")
```


```{r}
BPS_peaks <- FindMarkers(
  object = F2.Germ,
  ident.1 = 'S2',
  ident.2 = 'C2',
  only.pos = TRUE,
  test.use = 'LR',
  min.pct = 0.05,
  latent.vars = 'nFeature_ATAC'
)


top.da.peak <- rownames(BPS_peaks[BPS_peaks$p_val < 0.005, ])

enriched.motifs <- FindMotifs(
  object = F2.Germ,
  features = top.da.peak
)


write.csv(enriched.motifs, "F2.BPS.CON.enrichmotif.csv")
```

```{r}
saveRDS(F2.Germ, "F2.Germ.rds")
```


```{r}
MotifPlot(object = F2.Germ,
  motifs = head(rownames(enriched.motifs)),
  assay = 'ATAC')
ggsave("F2.BPS.enriched.motifs.tiff", dpi=300, width = 7, height = 4, bg="white")
```


```{r}
F2.Germ <- RunChromVAR(
  object = F2.Germ,
  genome = BSgenome.Mmusculus.UCSC.mm10
)

DefaultAssay(F2.Germ) <- 'chromvar'
```

```{r}
differential.activity <- FindMarkers(
  object = F2.Germ,
  ident.1 = 'A2',
  ident.2 = 'C2',
  only.pos = TRUE,
  mean.fxn = rowMeans,
  fc.name = "avg_diff"
)

write.csv(differential.activity,"F2.BPA.CON.diff_Motif_activity.csv")

differential.activity <- read.csv("F2.BPA.CON.diff_Motif_activity.csv")
enriched.motifs <- read.csv("F2.BPA.CON.enrichmotif.csv")
BPA.CON.DO <- merge(enriched.motifs, differential.activity,  by="motif")
write.csv(BPA.CON.DO, "F2.BPA.CON.diff_Motif_activity.combined.csv")
```


```{r}
differential.activity1 <- FindMarkers(
  object = F2.Germ,
  ident.1 = 'S2',
  ident.2 = 'C2',
  only.pos = TRUE,
  mean.fxn = rowMeans,
  fc.name = "avg_diff"
)

write.csv(differential.activity1,"F2.BPS.CON.diff_Motif_activity.csv")

differential.activity1 <- read.csv("F2.BPS.CON.diff_Motif_activity.csv")
enriched.motifs <- read.csv("F2.BPS.CON.enrichmotif.csv")
BPS.CON.DO <- merge(enriched.motifs, differential.activity1,  by="motif")
write.csv(BPS.CON.DO, "F2.BPS.CON.diff_Motif_activity.combined.csv")
```
```{r}
VlnPlot(F2.Germ, c(""))+NoLegend()+theme(axis.title.x = element_blank(), axis.title.y = element_text(size = 12))+theme(axis.text = element_text(color='black', size=12))+theme(legend.title = element_text(size=8)+theme(legend.text = element_text(size = 8)))+theme(axis.line.x = element_line(size=0.55))+theme(axis.line.y = element_line(size=0.55))+theme(axis.ticks.x = element_line(size=0.55))+theme(axis.ticks.y = element_line(size = 0.55))+theme(legend.title = element_text(size = 10))
ggsave("Etv4.tiff", dpi=300, width = 3, height = 2.5)
```

```{r}
#cluster5 GO analysis

library(clusterProfiler)
library(DOSE)
library(ggplot2)
library(enrichplot)
library(org.Mm.eg.db)
library(stringr)


cluster5 <- read.csv("F2.marker genes.csv")  
idcluster5 <- as.character(cluster5[,6])


cluster5$ENTREZID <- convertIDs(idcluster5, "SYMBOL","ENTREZID", org.Mm.eg.db)
head(cluster5, 4)
```

```{r}
F2.GO.BP <- enrichGO(cluster5$ENTREZID,   
                    OrgDb         = 'org.Mm.eg.db',
                    ont           = "BP",
                    pAdjustMethod = "BH",
                    pvalueCutoff  = 0.05,
                    qvalueCutoff  = 0.05,
                    readable      = TRUE)

write.csv(F2.GO.BP, "SSC5.GO.csv")
```




```{r}
a <- c("ribonucleoprotein complex biogenesis","cytoplasmic translation","translational elongation","positive regulation of signal transduction by p53 class mediator")

p1 <- barplot(F2.GO.BP, showCategory = a, font.size=10, x = "GeneRatio")+theme(axis.title = element_text(size=8))+theme(axis.text = element_text(color="black", size=8))+theme(legend.title = element_text(size=8)+theme(legend.text = element_text(size = 8)))+theme(legend.key.width = unit(0.2,"cm"))+theme(legend.key.height = unit(0.3,"cm"))+theme(legend.text = element_text(size = 6))+theme(legend.title = element_text(size = 6))+xlab("GeneRatio")+ggtitle("F2.SSC5.GO terms")+theme(plot.title = element_text(size=10,face = "bold"))+theme(axis.title = element_text(size = 8))+theme(axis.text.x = element_text(angle = 30, size=6))
p1


ggsave("F2.SSC5.GO.tiff", width=4, height=3, dpi=300, p1)
```



```{r}
#extract peaks position and make bed file to use Hormer for annotation
annotations <- GetGRangesFromEnsDb(ensdb = EnsDb.Mmusculus.v79)
seqlevelsStyle(annotations) <- "UCSC"
genome(annotations) <- "mm10"
Annotation(F2.Germ) <- annotations

DefaultAssay(F2.Germ) <- "ATAC"

peaks <- granges(F2.Germ)

df <- data.frame(seqnames=seqnames(peaks),
  starts=start(peaks)-1,
  ends=end(peaks),
  names=c(rep(".", length(peaks))),
  scores=c(rep(".", length(peaks))),
  strands=strand(peaks))

write.table(df, file="F2.germ.bed", quote=F, sep="\t", row.names=F, col.names=F)
```


```{r}
#merge homer annotation to BPA.BPS/CON DAR 
#in csv, change "Start" and "End" columns content type from general to number
df1<-read.csv("F2.BPA.CON.DAR.csv", head=T) 
df2<-read.csv("F2.homer.anno peaks.csv", head=T)


merge<-merge(df1,df2,by=c('Start','End'),all.x=T)

write.csv(merge, "F2.BPA.DAR.homer.anno.csv")



df1<-read.csv("F2.BPS.CON.DAR.csv", head=T)
df2<-read.csv("F2.homer.anno peaks.csv", head=T)


merge<-merge(df1,df2,by=c('Start','End'),all.x=T)

write.csv(merge, "F2.BPS.DAR.homer.anno.csv")


```



```{r}
DefaultAssay(F2.Germ) <-"RNA"
DotPlot_scCustom(F2.Germ, features = df, x_lab_rotate = TRUE, flip_axes = F, colors_use = colorRampPalette(c("#87cefa", "lightgrey", "#dc143c"))(256)) + xlab('Cluster') +  ylab('Genes')+theme(axis.title.x = element_blank(), axis.title.y = element_blank())+theme(axis.text = element_text(color='black', size=10))+theme(legend.key.size = unit(0.3, 'cm'))+theme(legend.title = element_text(size = 7))+theme(axis.line.x = element_line(size=0.55))+theme(axis.line.y = element_line(size=0.55))+theme(axis.ticks.x = element_line(size=0.55))+theme(axis.ticks.y = element_line(size = 0.55))+theme(legend.text = element_text(size = 6))

ggsave("F2.germ cell cluster markers1.tiff", dpi = 300, width = 8, height = 2.5, bg="white")
```

```{r}
saveRDS(F2.stem, "F2.stem.rds")
saveRDS(F2.Progenitor, "F2.Progenitor.rds")
saveRDS(F2.diff, "F2.diff.rds")
```

```{r}
library(monocle3)
library(SeuratWrappers)
library(cicero)






DefaultAssay(F2.Germ) <- "ATAC"
F2.germ.ATAC <- DietSeurat(F2.Germ, assays = "ATAC", dimreducs = "wnn.umap2Germ")

F2.germ.ATAC[["UMAP"]] <- F2.germ.ATAC[["wnn.umap2Germ"]]
F2.germ.ATAC.cds <- as.cell_data_set(F2.germ.ATAC)


F2.germ.ATAC.cicero <- make_cicero_cds(F2.germ.ATAC.cds, reduced_coordinates = reducedDims(F2.germ.ATAC.cds)$UMAP)
```


```{r}
library(BSgenome.Mmusculus.UCSC.mm10)

genome <- seqlengths(BSgenome.Mmusculus.UCSC.mm10)

genome.df <- data.frame("chr" = names(genome), "length" = genome)
# run cicero

conns <- run_cicero(F2.germ.ATAC.cicero,genomic_coords = genome.df, sample_num = 100) 
head(conns)
```

```{r}
#Find cis-co-accessible networks (CCANs)
ccans <- generate_ccans(conns)
head(ccans)
#Add links to Germ object
links <- ConnectionsToLinks(conns = conns, ccans = ccans)
Links(F2.Germ) <- links
```

```{r}
saveRDS(F2.Germ, "F2.Germ.rds")
```