
```{r}
df1<-read.csv("F1.BPS.up.csv", head=T)
df2<-read.csv("F2.BPS.up.csv", head=T)
df3<-read.csv("F3.BPS.up.csv", head=T)

BPA.up.overlap <- merge(x=df3, y=df1, by="Gene")
BPA.up.overlap <- merge(x=BPA.up.overlap, y=df2, by="Gene")
write.csv(BPA.up.overlap, "BPS.up.overlap.csv")
```


```{r}

df1<-read.csv("BPA.CON.DEs.F1.Germ.Ming.csv", head=T)
df2<-read.csv("BPS.CON.DEs.F1.Germ.Ming.csv", head=T)
df3<-read.csv("BPA.CON.DEs.F2.Germ.Ming.csv", head=T)
df4<-read.csv("BPS.CON.DEs.F2.Germ.Ming.csv", head=T)
df5<-read.csv("BPS.CON.DEs.F3.Germ.final.csv", head=T)
df6<-read.csv("BPA.CON.DEs.F3.Germ.final.csv", head=T)

library(dplyr)


overlap.ALL <- bind_rows(df1, df2, df3, df4, df5, df6) 
```







```{r}
df1<-read.csv("F1.BPS.down.csv", head=T)
df2<-read.csv("F2.BPS.down.csv", head=T)
df3<-read.csv("F3.BPS.down.csv", head=T)

BPA.down.overlap <- merge(x=df3, y=df1, by="Gene")
BPA.down.overlap <- merge(x=BPA.up.overlap, y=df2, by="Gene")
write.csv(BPA.down.overlap, "BPS.down.overlap.csv")
```



```{r}
library(clusterProfiler)
library(DOSE)
library(ggplot2)
library(enrichplot)
library(org.Mm.eg.db)
library(stringr)







A <- read.csv("BPA.up.overlap.csv")  
FoldChangePsDEs <- A[,2]
names(FoldChangePsDEs) <- as.character(A[,1])
FoldChangePsDEs <- sort(FoldChangePsDEs, decreasing = TRUE)

convertIDs <- function( ids, fromKey, toKey, db, ifMultiple=c( "putNA", "useFirst" ) ) {
   stopifnot( inherits( db, "AnnotationDb" ) )
   ifMultiple <- match.arg( ifMultiple )
   suppressWarnings( selRes <- AnnotationDbi::select( 
      db, keys=ids, keytype=fromKey, columns=c(fromKey,toKey) ) )
   if( ifMultiple == "putNA" ) {
      duplicatedIds <- selRes[ duplicated( selRes[,1] ), 1 ]   
      selRes <- selRes[ ! selRes[,1] %in% duplicatedIds, ] }
   return( selRes[ match( ids, selRes[,1] ), 2 ] )
}


idPsDEsA <- names(FoldChangePsDEs)
A$ENTREZID <- convertIDs(idPsDEsA, "SYMBOL","ENTREZID", org.Mm.eg.db)




S <- read.csv("BPS.up.overlap.csv")  
FoldChangePsDEs <- S[,2]
names(FoldChangePsDEs) <- as.character(S[,1])
FoldChangePsDEs <- sort(FoldChangePsDEs, decreasing = TRUE)


idPsDEsS <- names(FoldChangePsDEs)
S$ENTREZID <- convertIDs(idPsDEsS, "SYMBOL","ENTREZID", org.Mm.eg.db)



DATA <- list(BPA=A$ENTREZID, BPS=S$ENTREZID)

```



```{r}
AllDEs <- read.csv("All.csv")
FoldChangeAllDEs <- AllDEs[,2] 
names(FoldChangeAllDEs) <- as.character(AllDEs[,1])
FoldChangeAllDEs <- sort(FoldChangeAllDEs, decreasing = TRUE)

idAllDEs <- names(FoldChangeAllDEs)
AllDEs$ENTREZID <- convertIDs(idAllDEs, "SYMBOL","ENTREZID", org.Mm.eg.db )
head(AllDEs, 4)
```

```{r}
UP <- compareCluster(geneCluster = DATA, fun = enrichGO, ont = "BP",
                    pAdjustMethod = "BH",
                    pvalueCutoff  = 0.05,
                    qvalueCutoff  = 0.05,
                    universe= AllDEs$ENTREZID,
                    OrgDb = 'org.Mm.eg.db',
                    readable=TRUE)

head(UP) 

UP.filter <-simplify(UP, cutoff = 0.7, by = "p.adjust", select_fun = min)

#dim(ck.filter)

write.csv(UP.filter, "GO.overlap.UP.filter.csv")
```



```{r}
a <- c("ribonucleoprotein complex biogenesis","posttranscriptional regulation of gene expression","ribosome biogenesis","DNA repair","ATP metabolic process","oxidative phosphorylation","regulation of translation","meiotic cell cycle","meiosis I","methylation","autophagy","histone modification","mitotic cell cycle phase transition","RNA splicing","response to oxidative stress")

p1 <- dotplot(UP, showCategory = a, font.size=12, x="group")+facet_grid(~Cluster)+theme(axis.title = element_text(size=8))+theme(axis.text = element_text(color="black", size=8))+theme(legend.title = element_text(size=8)+theme(legend.text = element_text(size = 10)))+theme(legend.key.width = unit(0.2,"cm"))+theme(legend.key.height = unit(0.3,"cm"))+theme(legend.text = element_text(size = 8))+theme(legend.title = element_text(size = 8))+theme(plot.title = element_text(size=10,face = "bold"))+theme(axis.title = element_text(size = 8))+theme(axis.text.x = element_blank())+xlab("")
p1# facet_grid(~Cluster)+

ggsave("Go.OVERLAP.BP.cut1.tiff", dpi = 300, width = 4.2, height = 4)
```

```{r}
#Compare F1-F3 BPA vs BPS

library(clusterProfiler)
library(DOSE)
library(ggplot2)
library(enrichplot)
library(org.Mm.eg.db)
library(stringr)







F1 <- read.csv("F1.BPA.BPS.up.csv")  
FoldChangePsDEs <- F1[,2]
names(FoldChangePsDEs) <- as.character(F1[,1])
FoldChangePsDEs <- sort(FoldChangePsDEs, decreasing = TRUE)



idPsDEsF1 <- names(FoldChangePsDEs)
F1$ENTREZID <- convertIDs(idPsDEsF1, "SYMBOL","ENTREZID", org.Mm.eg.db)




F2 <- read.csv("F2.BPA.BPS.up.csv")  
FoldChangePsDEs <- F2[,2]
names(FoldChangePsDEs) <- as.character(F2[,1])
FoldChangePsDEs <- sort(FoldChangePsDEs, decreasing = TRUE)


idPsDEsF2 <- names(FoldChangePsDEs)
F2$ENTREZID <- convertIDs(idPsDEsF2, "SYMBOL","ENTREZID", org.Mm.eg.db)



F3 <- read.csv("F3.BPA.BPS.up.csv")  
FoldChangePsDEs <- F3[,2]
names(FoldChangePsDEs) <- as.character(F3[,1])
FoldChangePsDEs <- sort(FoldChangePsDEs, decreasing = TRUE)


idPsDEsF3 <- names(FoldChangePsDEs)
F3$ENTREZID <- convertIDs(idPsDEsF3, "SYMBOL","ENTREZID", org.Mm.eg.db)




DATA <- list(F1=F1$ENTREZID, F2=F2$ENTREZID, F3=F3$ENTREZID)
```




```{r}
UP <- compareCluster(geneCluster = DATA, fun = enrichGO, ont = "BP",
                    pAdjustMethod = "BH",
                    pvalueCutoff  = 0.05,
                    qvalueCutoff  = 0.05,
                    OrgDb = 'org.Mm.eg.db',
                    readable=TRUE)

head(UP) 

UP.filter <-simplify(UP, cutoff = 0.7, by = "p.adjust", select_fun = min)

#dim(ck.filter)

write.csv(UP.filter, "GO.BPA.BPS.UP.filter.csv")
```

```{r}
a <- c("ribonucleoprotein complex biogenesis","DNA repair","methylation","histone modification","cytoplasmic translation","cell cycle phase transition","ATP metabolic process","oxidative phosphorylation","regulation of translation","fertilization","regulation of response to oxidative stress")

p1 <- dotplot(UP.filter, showCategory = a, font.size=12, x="Count")+ facet_grid(~Cluster)+theme(axis.title = element_text(size=8))+theme(axis.text = element_text(color="black", size=8))+theme(legend.title = element_text(size=8)+theme(legend.text = element_text(size = 8)))+theme(legend.key.width = unit(0.2,"cm"))+theme(legend.key.height = unit(0.3,"cm"))+theme(legend.text = element_text(size = 6))+theme(legend.title = element_text(size = 6))+xlab("Gene Counts")+theme(plot.title = element_text(size=10,face = "bold"))+theme(axis.title = element_text(size = 8))+theme(axis.text.x = element_text(angle = 30, size=6))
p1
```

```{r}
#Compare F1-F3 BPA UP

library(clusterProfiler)
library(DOSE)
library(ggplot2)
library(enrichplot)
library(org.Mm.eg.db)
library(stringr)







F1 <- read.csv("F1.BPA.up.csv")  
FoldChangePsDEs <- F1[,2]
names(FoldChangePsDEs) <- as.character(F1[,1])
FoldChangePsDEs <- sort(FoldChangePsDEs, decreasing = TRUE)



idPsDEsF1 <- names(FoldChangePsDEs)
F1$ENTREZID <- convertIDs(idPsDEsF1, "SYMBOL","ENTREZID", org.Mm.eg.db)




F2 <- read.csv("F2.BPA.up.csv")  
FoldChangePsDEs <- F2[,2]
names(FoldChangePsDEs) <- as.character(F2[,1])
FoldChangePsDEs <- sort(FoldChangePsDEs, decreasing = TRUE)


idPsDEsF2 <- names(FoldChangePsDEs)
F2$ENTREZID <- convertIDs(idPsDEsF2, "SYMBOL","ENTREZID", org.Mm.eg.db)



F3 <- read.csv("F3.BPA.up.csv")  
FoldChangePsDEs <- F3[,2]
names(FoldChangePsDEs) <- as.character(F3[,1])
FoldChangePsDEs <- sort(FoldChangePsDEs, decreasing = TRUE)


idPsDEsF3 <- names(FoldChangePsDEs)
F3$ENTREZID <- convertIDs(idPsDEsF3, "SYMBOL","ENTREZID", org.Mm.eg.db)




DATA <- list(F1=F1$ENTREZID, F2=F2$ENTREZID, F3=F3$ENTREZID)
```

```{r}
up <- compareCluster(geneCluster = DATA, fun = enrichGO, ont = "BP",
                    pAdjustMethod = "BH",
                    pvalueCutoff  = 0.05,
                    qvalueCutoff  = 0.05,
                    OrgDb = 'org.Mm.eg.db',
                    readable=TRUE)

head(up) 

up.filter <-simplify(up, cutoff = 0.7, by = "p.adjust", select_fun = min)
write.csv(up, "F1-F3,GO.BPA.up.csv")
#dim(ck.filter)

write.csv(up.filter, "F1-F3.GO.BPA.up.filter.csv")
```

```{r}
a <- c("ribonucleoprotein complex biogenesis","RNA splicing","DNA repair","regulation of translation","mitotic cell cycle phase transition","ATP metabolic process","oxidative phosphorylation","intrinsic apoptotic signaling pathway","meiotic cell cycle process","autophagy","methylation","ATP metabolic process","histone modification","aerobic respiration","response to oxidative stress")

p1 <- dotplot(up, showCategory = a, font.size=10, x="Count")+ facet_grid(~Cluster)+theme(axis.title = element_text(size=8))+theme(axis.text = element_text(color="black", size=8))+theme(legend.title = element_text(size=8)+theme(legend.text = element_text(size = 8)))+theme(legend.key.width = unit(0.2,"cm"))+theme(legend.key.height = unit(0.3,"cm"))+theme(legend.text = element_text(size = 6))+theme(legend.title = element_text(size = 6))+xlab("Gene Counts")+theme(plot.title = element_text(size=10,face = "bold"))+theme(axis.title = element_text(size = 8))+theme(axis.text.x = element_text(angle = 30, size=6))+scale_color_gradientn(colors = c("#EF415E","#F282B4","#FDE4E3"))
p1

ggsave("F1-F3.GO.BPA.UP.tiff", dpi = 300, width = 6, height = 4)
```


```{r}
#Compare F1-F3 BPS UP

library(clusterProfiler)
library(DOSE)
library(ggplot2)
library(enrichplot)
library(org.Mm.eg.db)
library(stringr)







F1 <- read.csv("F1.BPS.up.csv")  
FoldChangePsDEs <- F1[,2]
names(FoldChangePsDEs) <- as.character(F1[,1])
FoldChangePsDEs <- sort(FoldChangePsDEs, decreasing = TRUE)



idPsDEsF1 <- names(FoldChangePsDEs)
F1$ENTREZID <- convertIDs(idPsDEsF1, "SYMBOL","ENTREZID", org.Mm.eg.db)




F2 <- read.csv("F2.BPS.up.csv")  
FoldChangePsDEs <- F2[,2]
names(FoldChangePsDEs) <- as.character(F2[,1])
FoldChangePsDEs <- sort(FoldChangePsDEs, decreasing = TRUE)


idPsDEsF2 <- names(FoldChangePsDEs)
F2$ENTREZID <- convertIDs(idPsDEsF2, "SYMBOL","ENTREZID", org.Mm.eg.db)



F3 <- read.csv("F3.BPS.up.csv")  
FoldChangePsDEs <- F3[,2]
names(FoldChangePsDEs) <- as.character(F3[,1])
FoldChangePsDEs <- sort(FoldChangePsDEs, decreasing = TRUE)


idPsDEsF3 <- names(FoldChangePsDEs)
F3$ENTREZID <- convertIDs(idPsDEsF3, "SYMBOL","ENTREZID", org.Mm.eg.db)




DATA <- list(F1=F1$ENTREZID, F2=F2$ENTREZID, F3=F3$ENTREZID)
```
```{r}
upS <- compareCluster(geneCluster = DATA, fun = enrichGO, ont = "BP",
                    pAdjustMethod = "BH",
                    pvalueCutoff  = 0.05,
                    qvalueCutoff  = 0.05,
                    OrgDb = 'org.Mm.eg.db',
                    readable=TRUE)

head(up) 

upS.filter <-simplify(upS, cutoff = 0.7, by = "p.adjust", select_fun = min)
write.csv(upS, "F1-F3,GO.BPS.up.csv")
#dim(ck.filter)

write.csv(upS.filter, "F1-F3.GO.BPS.up.filter.csv")
```





```{r}
a <- c("ribonucleoprotein complex biogenesis","RNA splicing","DNA repair","regulation of translation","mitotic cell cycle phase transition","ATP metabolic process","oxidative phosphorylation","intrinsic apoptotic signaling pathway","meiotic cell cycle process","autophagy","methylation","ATP metabolic process","histone modification","aerobic respiration","response to oxidative stress")

p2 <- dotplot(upS, showCategory = a, font.size=10, x="Count")+ facet_grid(~Cluster)+theme(axis.title = element_text(size=8))+theme(axis.text = element_text(color="black", size=8))+theme(legend.title = element_text(size=8)+theme(legend.text = element_text(size = 8)))+theme(legend.key.width = unit(0.2,"cm"))+theme(legend.key.height = unit(0.3,"cm"))+theme(legend.text = element_text(size = 6))+theme(legend.title = element_text(size = 6))+xlab("Gene Counts")+theme(plot.title = element_text(size=10,face = "bold"))+theme(axis.title = element_text(size = 8))+theme(axis.text.x = element_text(angle = 30, size=6))+scale_color_gradientn(colors = c("#0567A8","#7ED2DF","#D7EAF3"))
p2



ggsave("F1-F3.GO.BPS.UP.tiff", dpi = 300, width = 6, height = 4)
```



```{r}
F1.BPA <- read.csv("F1.BPA.up.csv")
F2.BPA <- read.csv("F2.BPA.up.csv")
F3.BPA <- read.csv("F3.BPA.up.csv")
F1.BPS <- read.csv("F1.BPS.up.csv")
F2.BPS <- read.csv("F2.BPS.up.csv")
F3.BPS <- read.csv("F3.BPS.up.csv")


x <- list(
 F1.BPA = as.matrix(F1.BPA[,1]), 
 F2.BPA  = as.matrix(F2.BPA[,1]),
 F3.BPA  = as.matrix(F3.BPA[,1]))

y <- list(
 F1.BPS = as.matrix(F1.BPS[,1]), 
 F2.BPS = as.matrix(F2.BPS[,1]),
 F3.BPS = as.matrix(F3.BPS[,1]))

library(ggVennDiagram)
library(ggvenn)
library(ggplot2)
ggvenn(
  x, 
  fill_color = c("#CD534CFF", "#EFC000FF","lightblue"),
  stroke_size = 0.3, set_name_size = 4
  )

ggsave("Venn plot.F1-F3.DEGs.BPA.up.tiff", dpi = 300, width = 6, height = 4)

ggvenn(
  y, 
  fill_color = c("#CD534CFF", "#EFC000FF","lightblue"),
  stroke_size = 0.3, set_name_size = 4
  )

ggsave("Venn plot.F1-F3.DEGs.BPS.up.tiff", dpi = 300, width = 6, height = 4)
```

```{r}
#Compare F1-F3 BPS down

library(clusterProfiler)
library(DOSE)
library(ggplot2)
library(enrichplot)
library(org.Mm.eg.db)
library(stringr)







F1 <- read.csv("F1.BPS.down.csv")  
FoldChangePsDEs <- F1[,2]
names(FoldChangePsDEs) <- as.character(F1[,1])
FoldChangePsDEs <- sort(FoldChangePsDEs, decreasing = TRUE)



idPsDEsF1 <- names(FoldChangePsDEs)
F1$ENTREZID <- convertIDs(idPsDEsF1, "SYMBOL","ENTREZID", org.Mm.eg.db)




F2 <- read.csv("F2.BPS.down.csv")  
FoldChangePsDEs <- F2[,2]
names(FoldChangePsDEs) <- as.character(F2[,1])
FoldChangePsDEs <- sort(FoldChangePsDEs, decreasing = TRUE)


idPsDEsF2 <- names(FoldChangePsDEs)
F2$ENTREZID <- convertIDs(idPsDEsF2, "SYMBOL","ENTREZID", org.Mm.eg.db)



F3 <- read.csv("F3.BPS.down.csv")  
FoldChangePsDEs <- F3[,2]
names(FoldChangePsDEs) <- as.character(F3[,1])
FoldChangePsDEs <- sort(FoldChangePsDEs, decreasing = TRUE)


idPsDEsF3 <- names(FoldChangePsDEs)
F3$ENTREZID <- convertIDs(idPsDEsF3, "SYMBOL","ENTREZID", org.Mm.eg.db)




DATA <- list(F1=F1$ENTREZID, F2=F2$ENTREZID, F3=F3$ENTREZID)
```
```{r}
downS <- compareCluster(geneCluster = DATA, fun = enrichGO, ont = "BP",
                    pAdjustMethod = "BH",
                    pvalueCutoff  = 0.05,
                    qvalueCutoff  = 0.05,
                    OrgDb = 'org.Mm.eg.db',
                    readable=TRUE)

head(downS) 

downS.filter <-simplify(downS, cutoff = 0.7, by = "p.adjust", select_fun = min)
write.csv(downS, "F1-F3,GO.BPS.down.csv")
#dim(ck.filter)

write.csv(downS.filter, "F1-F3.GO.BPS.down.filter.csv")
```


```{r}
b <- c("response to acid chemical","response to wounding","response to mechanical stimulus","response to amino acid","transmembrane receptor protein serine/threonine kinase signaling pathway","cellular response to glucose stimulus","cellular response to hormone stimulus","cellular response to hexose stimulus","response to calcium ion","cellular response to transforming growth factor beta stimulus","transmembrane receptor protein serine/threonine kinase signaling pathway")

p2 <- dotplot(downS, showCategory = a, font.size=10, x="Count")+ facet_grid(~Cluster)+theme(axis.title = element_text(size=8))+theme(axis.text = element_text(color="black", size=8))+theme(legend.title = element_text(size=8)+theme(legend.text = element_text(size = 8)))+theme(legend.key.width = unit(0.2,"cm"))+theme(legend.key.height = unit(0.3,"cm"))+theme(legend.text = element_text(size = 6))+theme(legend.title = element_text(size = 6))+xlab("Gene Counts")+theme(plot.title = element_text(size=10,face = "bold"))+theme(axis.title = element_text(size = 8))+theme(axis.text.x = element_text(angle = 30, size=6))+scale_color_gradientn(colors = c("#7b2cbf","#c77dff","#f2daff"))
p2



ggsave("F1-F3.GO.BPS.Down.tiff", dpi = 300, width = 6, height = 4)

```



```{r}
#Compare F1-F3 BPA down

library(clusterProfiler)
library(DOSE)
library(ggplot2)
library(enrichplot)
library(org.Mm.eg.db)
library(stringr)







F1 <- read.csv("F1.BPA.down.csv")  
FoldChangePsDEs <- F1[,2]
names(FoldChangePsDEs) <- as.character(F1[,1])
FoldChangePsDEs <- sort(FoldChangePsDEs, decreasing = TRUE)



idPsDEsF1 <- names(FoldChangePsDEs)
F1$ENTREZID <- convertIDs(idPsDEsF1, "SYMBOL","ENTREZID", org.Mm.eg.db)




F2 <- read.csv("F2.BPA.down.csv")  
FoldChangePsDEs <- F2[,2]
names(FoldChangePsDEs) <- as.character(F2[,1])
FoldChangePsDEs <- sort(FoldChangePsDEs, decreasing = TRUE)


idPsDEsF2 <- names(FoldChangePsDEs)
F2$ENTREZID <- convertIDs(idPsDEsF2, "SYMBOL","ENTREZID", org.Mm.eg.db)



F3 <- read.csv("F3.BPA.down.csv")  
FoldChangePsDEs <- F3[,2]
names(FoldChangePsDEs) <- as.character(F3[,1])
FoldChangePsDEs <- sort(FoldChangePsDEs, decreasing = TRUE)


idPsDEsF3 <- names(FoldChangePsDEs)
F3$ENTREZID <- convertIDs(idPsDEsF3, "SYMBOL","ENTREZID", org.Mm.eg.db)




DATA <- list(F1=F1$ENTREZID, F2=F2$ENTREZID, F3=F3$ENTREZID)
```

```{r}
down <- compareCluster(geneCluster = DATA, fun = enrichGO, ont = "BP",
                    pAdjustMethod = "BH",
                    pvalueCutoff  = 0.05,
                    qvalueCutoff  = 0.05,
                    OrgDb = 'org.Mm.eg.db',
                    readable=TRUE)

head(down) 

down.filter <-simplify(down, cutoff = 0.7, by = "p.adjust", select_fun = min)
write.csv(down, "F1-F3,GO.BPA.down.csv")
#dim(ck.filter)

write.csv(down.filter, "F1-F3.GO.BPA.down.filter.csv")
```


```{r}
b <- c("response to peptide","response to wounding","response to mechanical stimulus","response to amino acid","transmembrane receptor protein serine/threonine kinase signaling pathway","cellular response to glucose stimulus","cellular response to hormone stimulus","cellular response to hexose stimulus","response to calcium ion","cellular response to transforming growth factor beta stimulus","methylation","histone modification","leukocyte migration")

p2 <- dotplot(down, showCategory = b, font.size=8, x="Count")+ facet_grid(~Cluster)+theme(axis.title = element_text(size=8))+theme(axis.text = element_text(color="black", size=8))+theme(legend.title = element_text(size=8)+theme(legend.text = element_text(size = 8)))+theme(legend.key.width = unit(0.2,"cm"))+theme(legend.key.height = unit(0.3,"cm"))+theme(legend.text = element_text(size = 6))+theme(legend.title = element_text(size = 6))+xlab("Gene Counts")+theme(plot.title = element_text(size=10,face = "bold"))+theme(axis.title = element_text(size = 8))+theme(axis.text.x = element_text(angle = 30, size=6))
p2

ggsave("F1-F3.GO.BPA.Down.tiff", dpi = 300, width = 6, height = 4)
```




```{r}
library("dplyr")   
df1 <- read.csv("BPA.up.overlap.csv", header = T)
df2 <- read.csv("BPS.up.overlap.csv", header = T)


diff <-  anti_join(x=df1, y=df2, by = join_by(Gene == Gene))


write.csv(diff,"F1-F3.BPA.overlap.up diff.csv")


merge <- merge(x=df2, y=df1, by="Gene")
write.csv(merge,"F1-F3 up.BPA.BPS.merge.csv")
```


```{r}
x <- list(
  A.up = as.matrix(df1[,1]), 
  
 S.up = as.matrix(df2[,1])
  
    )


library(ggVennDiagram)
library(ggvenn)
library(ggplot2)
ggvenn(x)
```


```{r}

library("dplyr")   
df1 <- read.csv("F1-F3.BPS.overlap.up diff.csv", header = T)
df2 <- read.csv("F1-F3 overlap GO S specific genes.csv", header = T)

merge <- merge(x=df2, y=df1, by="Gene")
write.csv(merge,"F1-F3 overlap Go.BPS.specific genes merge with F1-F3 BPS overlap up diff.csv")
```


```{r}
library("dplyr")   
df1 <- read.csv("A.GO.overlap.UP.csv", header = T)
df2 <- read.csv("S.GO.overlap.UP.csv", header = T)


diff <-  anti_join(x=df2, y=df1, by = join_by(ID == ID))


write.csv(diff,"F1-F3.GO.overlap.up S diff.csv")

```


```{r}
df <-read.csv("F1-F3 overlap GO S specific genes.csv")
df <-df[,2]
Idents(F1.Germ) <- "treatment"
levels(F1.Germ)
library(circlize)
library(RColorBrewer)
DotPlot_scCustom(F1.Germ, features = df, x_lab_rotate = TRUE, flip_axes = F,colors_use = colorRampPalette(c("#87cefa", "lightgrey", "#dc143c"))(256)) + xlab('Cluster') +  ylab('Genes')+theme(axis.title.x = element_blank(), axis.title.y = element_blank())+theme(axis.text = element_text(color='black', size=7))+theme(legend.key.size = unit(0.3, 'cm'))+theme(legend.title = element_text(size = 7))+theme(axis.line.x = element_line(size=0.55))+theme(axis.line.y = element_line(size=0.55))+theme(axis.ticks.x = element_line(size=0.55))+theme(axis.ticks.y = element_line(size = 0.55))+theme(legend.text = element_text(size = 6))
ggsave("Expression of genes involved in BPS specific upregulated GO terms in F1.tiff", dpi = 300, width = 8, height = 2, bg="white")
DotPlot_scCustom(F2.Germ, features = df, x_lab_rotate = TRUE, flip_axes = F,colors_use = colorRampPalette(c("#87cefa", "lightgrey", "#dc143c"))(256)) + xlab('Cluster') +  ylab('Genes')+theme(axis.title.x = element_blank(), axis.title.y = element_blank())+theme(axis.text = element_text(color='black', size=7))+theme(legend.key.size = unit(0.3, 'cm'))+theme(legend.title = element_text(size = 7))+theme(axis.line.x = element_line(size=0.55))+theme(axis.line.y = element_line(size=0.55))+theme(axis.ticks.x = element_line(size=0.55))+theme(axis.ticks.y = element_line(size = 0.55))+theme(legend.text = element_text(size = 6))
ggsave("Expression of genes involved in BPS specific upregulated GO terms in F2.tiff", dpi = 300, width = 8, height = 2, bg="white")
DotPlot_scCustom(F3.Germ, features = df, x_lab_rotate = TRUE, flip_axes = F,colors_use = colorRampPalette(c("#87cefa", "lightgrey", "#dc143c"))(256)) + xlab('Cluster') +  ylab('Genes')+theme(axis.title.x = element_blank(), axis.title.y = element_blank())+theme(axis.text = element_text(color='black', size=7))+theme(legend.key.size = unit(0.3, 'cm'))+theme(legend.title = element_text(size = 7))+theme(axis.line.x = element_line(size=0.55))+theme(axis.line.y = element_line(size=0.55))+theme(axis.ticks.x = element_line(size=0.55))+theme(axis.ticks.y = element_line(size = 0.55))+theme(legend.text = element_text(size = 6))
ggsave("Expression of genes involved in BPS specific upregulated GO terms in F3.tiff", dpi = 300, width = 8, height = 2, bg="white")
```
```{r}
df4 <- AverageExpression(
  F3.Germ,
  assays = "RNA",
  features = df,
  return.seurat = FALSE,
  group.by = "Treatment",
  add.ident = NULL,
  slot = "data",
  verbose = TRUE
)

write.csv(df4, "BPS f1-f3 overlap GO specific gene heatmap.csv")
```

```{r}
 df6 <- read.csv("BPS f1-f3 overlap GO specific gene heatmap.csv")
 myCol <- colorRampPalette(c("blue", "white", "red"))(256)
 
 rownames(df6) <- df6$Gene
gsn <- grep("Gene", colnames(df6))
df6 <- df6[,-gsn]



group_df = data.frame(Groups=as.factor(c("CON", "BPA","BPS")))
rownames(group_df) <- colnames(df6)
ann_colors = list(
  Groups = c(CON="#98AFC7", BPA="#F8766D",BPS="#FDBD01"))


library(RColorBrewer)
library(circlize)
library(ComplexHeatmap)
library(ggplot2)
library(dplyr)


png(file="BPS f1-f3 overlap GO specific gene heatmap.png", width=3,height=6.0,units="in",res=1200)

library("pheatmap")
my_heatmap<-pheatmap(as.matrix(df6), scale="row", color=colorRampPalette(c("navy", "white", "red"))(50), 
          annotation_col = group_df, annotation_colors = ann_colors, cluster_rows= FALSE, cluster_cols = FALSE, cellwidth=40, cellheight=5, show_colnames = F, angle_col = c("0"), annotation_legend=F, fontsize_row =5, legend_labels = c("-2","-1","0","1","2"), name =  "Key",fontsize = 2)

draw(my_heatmap)
 dev.off()
```
```{r}
# Compare BPA vs BPS
df <- read.csv("F1-F3 overlapped UP DEGs.csv")  
FoldChange <- df[,2]
names(FoldChange) <- as.character(df[,1])
treatment<-df[,3]

id <- names(FoldChange)
df$ENTREZID <- convertIDs(id, "SYMBOL","ENTREZID", org.Mm.eg.db)


mydf <- data.frame(Entrez=df$ENTREZID, FC=FoldChange, Treatment=treatment)

mydf$group[mydf$Treatment == "A"] <- "BPA"
mydf$group[mydf$Treatment == "S"] <- "BPS"
```


```{r}
BPA.S <- compareCluster(Entrez~group, data=mydf,fun = enrichGO, ont = "BP",
                    pAdjustMethod = "BH",
                    pvalueCutoff  = 0.05,
                    qvalueCutoff  = 0.05,
                    universe= AllDEs$ENTREZID,
                    OrgDb = 'org.Mm.eg.db',
                    readable=TRUE)

head(BPA.S) 
write.csv(BPA.S, "F1-F3 GO.BPA.S.csv")
```



