

```{r}
library(Seurat)
library(Signac)
library(EnsDb.Mmusculus.v79)    
library(BSgenome.Mmusculus.UCSC.mm10)
library(ggplot2)
library(future)
set.seed(1234)
library(dplyr)
library(hdf5r)
```



```{r}
DefaultAssay(F3.Germ)<-"ATAC"
```


```{r}
#1 Find the positions of each promoter

library(EnsDb.Mmusculus.v79)

gene.ranges <- genes(EnsDb.Mmusculus.v79)
gene.ranges <- gene.ranges[gene.ranges$gene_biotype == 'protein_coding', ]
gene.ranges <- keepStandardChromosomes(gene.ranges, pruning.mode = 'coarse')
seqlevelsStyle(gene.ranges) <- 'UCSC'
gene.promoters <- promoters(gene.ranges)
```



```{r}
#2 Find which promoters overlap a peak

library(GenomicRanges)
library(S4Vectors)

peaks <- StringToGRanges(regions = rownames(F3.Germ), sep = c(":", "-"))
hits <- findOverlaps(query = peaks, subject = gene.promoters)
promoter.peaks <- queryHits(hits)
write.csv(promoter.peaks, "promoter.peaks.csv")
```



```{r}
#3 Find which promoter peaks contain the motif

promoter.peaks <- read.csv("promoter.peaks.csv")
promoter.peaks <- promoter.peaks[,2]




promoter.peaks.with.motif <- GetMotifData(object = F3.Germ)[promoter.peaks, "MA1603.1"]

write.csv(promoter.peaks.with.motif, "peak_motif.BPA_overlap.csv")





promoter.peaks.with.motif <- GetMotifData(object = F3.Germ)[promoter.peaks, "MA1603.1"]

write.csv(promoter.peaks.with.motif, "peak_motif.BPS_overlap.csv")
```




```{r}
ATAC.peaks <- ClosestFeature(
  object = F3.Germ,
  regions = peaks
)
write.csv(ATAC.peaks, "F3.Germ.peaks.total.closestfeature.csv")
```



```{r}
# promoter peaks annotation
# add a new column to "F3.Germ.peaks.total.closestfeature.csv", head="x", content=number order, see the file.



df1<-read.csv("promoter.peaks.csv", head=T)
df2<-read.csv("F3.Germ.peaks.total.closestfeature.csv", head=T)


promoter.peak.anno <- merge(x=df1, y=df2, by="x")

#remove rows with na

#BPA.S.overlap <- BPA.S.overlap[complete.cases(BPA.S.overlap), ] 

#keep only has na
#BPA.S.overlap <- combine_list[is.na(BPA.S.overlap$pct.1.y), ]


write.csv(promoter.peak.anno, "promoter.peak.anno.csv")
#copy "query_region" column to "peak_motif.BPA_overlap.csv", and "peak_motif.BPS_overlap.csv"

```





```{r}
#Combine with DAR
df3<-read.csv("peak_motif.BPA_overlap.csv", head=T)
df4<-read.csv("F3.BPA.CON.DAR.csv", head=T)


BPA.overlap <- merge(x=df4, y=df3, by="query_region")
write.csv(BPA.overlap, "BPA.diff_motif_promoter_anno_DAR.csv")


df3<-read.csv("peak_motif.BPS_overlap.csv", head=T)
df4<-read.csv("F3.BPS.CON.DAR.csv", head=T)


BPS.overlap <- merge(x=df4, y=df3, by="query_region")
write.csv(BPS.overlap, "BPS.diff_motif_promoter_anno_DAR.csv")
```



```{r}
#select FC>=2, plot heatmap for TF motif activity and TF gene expression

#In F3, only Dmrt1
```




